src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,157 @@
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  // Check on label parameters.
+  // Check for punctuation.
+  RequireParamValue<int>("punctuation",
+      [](double x) { return x == 0 || x <= 1; }, true,
+      "value for punctuation may be either 0 to not\
+       remove punctuation or 1 to remove punctuation 1");
+  RequireParamValue<int>("lowercase",
+      [](double x) { return x == 0 || x <= 1; }, true,
+      "value for lowercase may be either 0 to not\
+       remove punctuation or 1 to remove punctuation 1");
+  RequireParamValue<int>("stopwords",
+      [](double x) { return x == 0 || x <= 1; }, true,
+      "value for stopwords may be either 0 to not\
+       remove punctuation or 1 to remove punctuation 1");
+  // If puntucation is not set, warn the user.
+  if (!CLI::HasParam("punctuation"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("punctuation")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // If lowercase is not set, warn the user.
+  if (!CLI::HasParam("lowercase"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("lowercase")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // If stopwords is not set, warn the user.
+  if (!CLI::HasParam("stopwords"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("stopwords")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // If dimension is not set, warn the user.
+  if (!CLI::HasParam("dimension"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("dimension")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // Ectracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // Extracting the Contents of file
+  // File pointer
+  fstream fin;
+  // Open an existing file
+  fin.open(filename, ios::in);
+  std::string line;
+  std::vector<std::string> temp;
+  std::vector<std::string> input;
+  std::vector<std::vector<std::string>> dataset;
+  const size_t Column = CLI::GetParam<int>("dimension");
+  while (std::getline(fin, line))
+  {
+    boost::split(temp, line, boost::is_any_of(std::string(1, ',')));
+    input.push_back(temp[Column]);
+    dataset.push_back(temp);
+    temp.clear();
+  }
+  fin.close();
+  if (CLI::GetParam<int>("lowercase"))
+  {
+    data::LowerCase(input);
+  }
+  if (CLI::GetParam<int>("stopwords"))
+  {
+    // Not sure how to take input for tokenizer from cli.
+    // data::RemoveStopWords(input);
+  }
+  if (CLI::GetParam<int>("punctuation"))
+  {
+    data::RemovePunctuation(input);
+  }
+  const std::string filename2 = CLI::GetParam<std::string>("preprocess"
+      "_dataset");
+  fstream fout;
+  fout.open(filename2, ios::out | ios::trunc);
+  for (size_t i = 0 ; i < dataset.size(); i++)
+  {
+    for (size_t j =0 ; j < dataset[i].size(); j++)
+    {
+      if (j == Column)
```

Comment:
Could you elaborate a bit, what is the purpose of the conditional jump below?

Reply:
Suppose our dataset is a csv file with the following data
```
10,100,Hello world,15
12,500,Machine Learning,20
```

Now we have to process strings, in the following dataset and hence i am taking an input from user for which of the above column i have to do string processing and then separating the column of numerical dataset into one vector and , the string part to the other.

So we would have follwing two vectors
```
10,100,15
12,500,20
```

```
Hello world
Machine Learning
```
I will do the functionalities on the second one, while remerging i have to mind that it should be placed at correct location and hence conditional jump.

Reply:
I think it is better to treat all data as strings since strings can contain digits.

Reply:
Umm, If you take a close look over the code , I have taken both as strings, I have just used two vectors, such that user may not always want to do processing at every column. 
For instance take a case :

```
1,10, Hello world.,45.6
3,0,Mlpack i's great?,53.6
```
Now for the above case , user might want to remove punctuation from string, but if we pass whole vector, than it will also transform 45.6 -> 456 and 53.6 ->536, along with other strings since `.` is considered as punctuation and that totally changes its meaning, hence i will take 1st, 2nd and 4th column in vector and 3rd in another , and pass the vector which has only 3rd column, to perform function and then merge them back. 

In my code, I have used the following two vectors :

```
std::vector<std::string> input; // Store data of a particular column to perform function upon
std::vector<std::vector<std::string>> dataset; // rest all columns
```

Reply:
Thanks for the clarification. But what is the sense of splitting a string which has already been split by commas?

Reply:
How should I divide it into two vectors then, I am just separating a particular column. Which I need to pass it to the function. While reading the rows, they are just a single string separated by `,`. And hence I am splitting it to store in`vector<vector<string>>`.


Or should I pass the whole dataset at once to the function for preprocessing.? Or you mean instead of story `vector<vector<string>>` store it in `vector<string>`